src/main/java/org/example/homework9/MobilePhone.java

```diff
@@ -0,0 +1,66 @@
+package org.example.homework9;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class MobilePhone {
+    List<Contact> myContacts;
+    String myNumber;
+
+    public MobilePhone(String myNumber) {
+        this.myNumber = myNumber;
+        this.myContacts = new ArrayList<>();
+    }
+
+    public boolean addNewContact(Contact contactToAdd) {
+        if (findContact(contactToAdd) >= 0) {
+            return false;
+        }
+        myContacts.add(contactToAdd);
+        return true;
+    }
+
+    public boolean updateContact(Contact oldContact, Contact newContact) {
+        if (findContact(oldContact) < 0) {
+            return false;
+        }
+        myContacts.remove(oldContact);
+        myContacts.add(newContact);
+        return true;
+    }
+
+    public boolean removeContact(Contact contactToRemove) {
+        return myContacts.remove(contactToRemove);
+    }
+
+    public int findContact(Contact contactToFind) {
+        return myContacts.indexOf(contactToFind);
+    }
+
+    public int findContact(String name) {
+        for (int i = 0; i < myContacts.size(); i++) {
+            if (myContacts.get(i).getName().equals(name)) {
+                return i;
+            }
+        }
+        return -1;
+    }
+
+    public Contact queryContact(String name) {
+        for (Contact contact : myContacts) {
+            if (contact.getName().equals(name)) {
+                return contact;
+            }
+        }
+        return null;
+    }
+
+    public void printContacts() {
+        StringBuilder contactsInfo = new StringBuilder("Contact List:").append(System.lineSeparator());
+        for (int i = 0; i < myContacts.size(); i++) {
+            Contact contact = myContacts.get(i);
+            contactsInfo.append(i + 1).append(". ").append(contact).append(System.lineSeparator());
```

Comment:
Using StringBuilder here is not the best option. You do not perform multiple appending here. 
It looks good in terms of learning, but to complicated for this particular case
You can overrride toString in Contact, to simplify it and have custom logic.
or just use someting like this here:
`System.out.println((i + 1) + ". " + myContacts.get(i).getName() + " -> " + myContacts.get(i).getPhoneNumber());`

Reply:
You can leave it as it is. But it looks a bit complicated for current logic. 
I agree that we produce the same amount of strings, but you do not do huge amount of string modification, and StringBuilder here is not mandatory(in terms of memory consumption)

Reply:
Ok, thank you. Made it.
If everything is ok, I will resubmit homework in LMS, ok?

Reply:
Sure, I approved PR, feel free to resubmit